runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7281,7 +7281,12 @@ done:;
 				&& J9_IS_J9CLASS_VALUETYPE(lhsClass))
 				&& (rhsClass == lhsClass)
 			) {
-				acmpResult = isSubstitutable(lhs, rhs);
+				if (J9_ARE_ALL_BITS_SET(lhsClass->classFlags, J9ClassCanSupportFastSubstitutability)) {
+					UDATA const objectHeaderSize = J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread);
+					acmpResult = (0 == memcmp((void *)((UDATA)lhs + objectHeaderSize), (void *)((UDATA)rhs + objectHeaderSize), lhsClass->totalInstanceSize));
```

Comment:
also use `J9_VALUETYPE_FLATTENED_SIZE` instead of `lhsClass->totalInstanceSize)`